Extra Logging

srleach opened the pull request:
## Added
- Extra logging to assist diagnosis of bug (#150323279)

wnowicki commented on app/Http/Middleware/Download.php:
```diff
@@ -37,6 +39,10 @@ public function handle($request, Closure $next)
     {
         $response = $next($request);
 
+        if (!$response instanceof Response) {
```
- use `[]` for each variable please don't introduce any new standards like `:` as then we can mixed our `:` with real `:`
- I'd dump even more and use *context* for that, if it's an exceptional case we want to see everything we can

```php
Log::warning('Unexpected response type: [' . get_class($response) . '] on route [' . $request->method() . '] [' . $request->path() . ']', ['response' => print_r($response, true), 'request' => print_r($request, true)]);
```

srleach commented:
@wnowicki I implemented *some* of your changes. I didn't want to dump the entire request as we would then capture credentials which I don't want in the logs. 

- I've separated the variables as requested
- I've added the request headers to the context
- Response is of unknown type, and for some reason an `Illuminate\Http\Response` will not `print_r()` so I've omitted this as we just want some quick visibility.

wnowicki commented:
@srleach 
- maybe I'm wrong by `print_r()` is able to deal with everything in PHP, and I think there is massive chance that we have to go back and implement it anyway to understand this case, so please try to put there `print_r()` or you need to prove me I'm wrong in what I'm saying :)

wnowicki commented:
- explanation that object is potentially t0o big is absolutely acceptable... I'd approve it if you would put it in the comment
- but then if size is too big then we have our candidate of an actual issue I'd say
- so happy to remove `print_r()` for now but I'd say it requires more investigation of this size issue before we can close this ticket

srleach commented:
Per our discussion, I do think size is the issue here. Further, I do not see it blocking closing this ticket as the purpose of the ticket is the identification of a bad route, this is a separate thing. 

There are no logs - the request simply processes and never returns when trying to print_r this response. I would expect this is because we're sending the entire response payload into the print_r function which will likely have heavy nesting and large member variables.

If we end up needing to see this response, then what we have now will allow us to log this in a cleaner manner. We'll see which route, headers and method caused the response to return with the incorrect type which will, in turn allow us to trace the root cause. 

Fairly interesting that print_r fails though!

srleach commented:
@wnowicki
